ci-operator/step-registry/ipi/conf/azure/ipi-conf-azure-commands.sh

```diff
@@ -20,41 +20,131 @@ function getVersion() {
   echo "${version}"
 }
 
+function join_by { local IFS="$1"; shift; echo "$*"; }
+
+# BootstrapInstanceType gets its value from pkg/types/azure/defaults/platform.go
+architecture=${OCP_ARCH:-"amd64"}
+if [[ "${CLUSTER_TYPE}" == "azure-arm64" ]]; then
+  architecture="arm64"
+fi
+
 CONFIG="${SHARED_DIR}/install-config.yaml"
 
 REGION="${LEASED_RESOURCE}"
 echo "Azure region: ${REGION}"
 
+if [[ "${COMPUTE_NODE_TYPE}" == "" ]]; then
+  if [ "${architecture}" = "arm64" ]; then
+    COMPUTE_NODE_TYPE="Standard_D4ps_v5"
+  else
+    COMPUTE_NODE_TYPE="Standard_D4s_v3"
+  fi
+fi
 workers=3
 if [[ "${SIZE_VARIANT}" == "compact" ]]; then
   workers=0
 fi
 master_type=null
-if [[ "${SIZE_VARIANT}" == "xlarge" ]]; then
-  master_type=Standard_D32s_v3
-elif [[ "${SIZE_VARIANT}" == "large" ]]; then
-  master_type=Standard_D16s_v3
-elif [[ "${SIZE_VARIANT}" == "compact" ]]; then
-  master_type=Standard_D8s_v3
+if [ "${architecture}" = "arm64" ]; then
+  BOOTSTRAP_NODE_TYPE="Standard_D8ps_v5"
+  if [[ "${SIZE_VARIANT}" == "xlarge" ]]; then
+    master_type="Standard_D32ps_v5"
+  elif [[ "${SIZE_VARIANT}" == "large" ]]; then
+    master_type="Standard_D16ps_v5"
+  elif [[ "${SIZE_VARIANT}" == "compact" ]]; then
+    master_type="Standard_D8ps_v5"
+  fi
+else
+  BOOTSTRAP_NODE_TYPE="Standard_D8s_v3"
+  if [[ "${SIZE_VARIANT}" == "xlarge" ]]; then
+    master_type="Standard_D32s_v3"
+  elif [[ "${SIZE_VARIANT}" == "large" ]]; then
+    master_type="Standard_D16s_v3"
+  elif [[ "${SIZE_VARIANT}" == "compact" ]]; then
+    master_type="Standard_D8s_v3"
+  fi
+fi
+
+echo "Using control plane instance type: ${master_type}"
+echo "Using compute instance type: ${COMPUTE_NODE_TYPE}"
+
+# set the parameters we'll need as env vars
```

Comment:
The purpose to add "az login" is that we need to run `az vm list-skus ...`  command， discussed with @MayXuQQ ，amd64 doesn't hit the zone limit issues for amd64, so we can keep the origin code without manually changing zone setting for amd64 and doesn't do existing_zone_setting check and keep it as it is.

We can only do it when arch is arm64 and supported zone are not 1,2,3(such as only avaliable in zone 1 or zone 1,2),  is it feasible for you?

Reply:
Given that we are doing it even for aws, I think we should have the same behavior for Azure and regardless of the architecture.  And I think what you've done is ok, it can be just re-sorted as suggested (or better).

However, I'm still wondering whether the installer can implement this feature. Users aiming to install a cluster in a region that has not the wanted machines in the AZs, don't need to check themselves and set the zones in the install-config.yaml. The installer would perform this kind of work to understand which zones can be used. Maybe @jianlinliu could help answer.